lib/screenplay/config/s3_fetch.ex

```diff
@@ -37,9 +37,15 @@ defmodule Screenplay.Config.S3Fetch do
 
   @impl true
```

Comment:
We should update the spec for this function in the `Screenplay.Config.Fetch` behaviour — it was specced to accept a `String.t()`, which was sort-of correct for the old version of the S3 function but not for the local function. I'm not sure why Dialyzer accepted this (a phrase I find myself using a lot), but it might be down to a lack of explicit types elsewhere in this code path.

It looks like this accepts an argument of the type `get_places_and_screens` returns in its `:ok` case, which is... currently specced as `term` 🙃 I know ultimately we could get back anything from decoding JSON, but is there an easy way we could clarify this type? Even if it's just "list of maps" or something.

Reply:
Huh, thanks for calling that out. I leaned too heavily on Dialyzer 😬 

As for the `term` spec, that's due to the face that `Jason.decode/1` is specced  to return `{:ok, term()}` so Dialyzer sees that and rejects `map()` in the spec. Have you solved something like that before?

Reply:
Nevermind... I think this was a problem with a spec higher up.